test/MatrixFields/scalar_fieldmatrix.jl

```diff
@@ -0,0 +1,273 @@
+using Test
+using JET
+
+import ClimaCore: Geometry, Domains, Meshes, Spaces, Fields, MatrixFields
+import ClimaCore.Utilities: half
+import ClimaCore.RecursiveApply: ⊠
+import ClimaComms
+import LinearAlgebra: I, norm, ldiv!, mul!
+import ClimaCore.MatrixFields: @name
+ClimaComms.@import_required_backends
+include("matrix_field_test_utils.jl")
+
+@testset "get_field_first_index_offset" begin
+    struct singleton{T}
```

Comment:
```suggestion
    struct Singleton{T}
```
Types should generally be capitalized. Same for `TwoFields` below.